### PR TITLE
Decimal precision fix

### LINKED ISSUE #1639: decimal precission issue in graphql
I am using GraphQL nuget package in one of my project. The version is 2.4.0.

When I am passing the decimal value as 12.10 the graphql is somehow transforming that into 12.1
If i send something like 12.11 then its coming as 12.11 but not when the last digit is zero. Is there something that i need to change in my code to get it working?

GraphQL decleration

`            Field(x => x.percentage, nullable: true).Name("percentage");`

Decleration

`public decimal? percentage{ get; set; }`

### PR DESCRIPTION
**Pull request by sungam3r**
fixes #1639

### CODE REVIEW

File: src/GraphQL/Language/CoreToVanillaConverter.cs [hunk 1]
```diff
@@ -231,7 +231,18 @@ public IValue Value(GraphQLValue source)
                 case ASTNodeKind.FloatValue:
                 {
                     var str = (GraphQLScalarValue)source;
-                    return new FloatValue(ValueConverter.ConvertTo<double>(str.Value)).WithLocation(str, _body);
+
+                    // the idea is to see if there is a loss of accuracy of value
+                    // for example, 12.1 or 12.11 is double but 12.10 is decimal
+                    double dbl = ValueConverter.ConvertTo<double>(str.Value);
+                    decimal dec = ValueConverter.ConvertTo<decimal>(str.Value);
+                    // TODO: make more efficient, current solution allocates memory
```
Comment (sungam3r) [inline src/GraphQL/Language/CoreToVanillaConverter.cs:239]: I will be glad if someone suggests a more efficient way.

File: src/GraphQL/Language/CoreToVanillaConverter.cs [hunk 2]
```diff
@@ -231,7 +231,18 @@ public IValue Value(GraphQLValue source)
...
+                    int[] decBits = decimal.GetBits(dec);
+                    int[] dblAsDecBits = decimal.GetBits(new decimal(dbl));
+                    if (decBits[0] != dblAsDecBits[0] || decBits[1] != dblAsDecBits[1] || decBits[2] != dblAsDecBits[2] || decBits[3] != dblAsDecBits[3])
+                        return new DecimalValue(dec);
```
Comment (SimonCropp) [inline src/GraphQL/Language/CoreToVanillaConverter.cs:243]: should there be a withlocation here?

File: src/GraphQL/Language/CoreToVanillaConverter.cs (hunk 1 above)
Comment (SimonCropp) [reply src/GraphQL/Language/CoreToVanillaConverter.cs:239]: happy to go with this for now

File: src/GraphQL/Language/CoreToVanillaConverter.cs (hunk 2 above)
Comment (sungam3r) [reply src/GraphQL/Language/CoreToVanillaConverter.cs:243]: 🤦 Of course, fixed.

**Pull request merged**